arch/risc-v/include/syscall.h

```diff
@@ -78,6 +79,7 @@
  */
 
 #define SYS_switch_context        (2)
+#endif /* CONFIG_ARCH_USE_S_MODE */
```

Comment:
need move after line 91?

Reply:
But the implementation of SYS_syscall_return is excluded by CONFIG_ARCH_USE_S_MODE?

Reply:
It is included, riscv_swint.c:
```

#endif /* CONFIG_ARCH_USE_S_MODE */


      /* A0=SYS_syscall_return: This is a SYSCALL return command:
      /* A0=SYS_syscall_return: This is a SYSCALL return command:
       *
       *
       *   void up_sycall_return(void);
       *   void up_sycall_return(void);
       *
       *
       * At this point, the following values are saved in context:
       * At this point, the following values are saved in context:
       *
       *
       *   A0 = SYS_syscall_return
       *   A0 = SYS_syscall_return
       *
       *
       * We need to restore the saved return address and return in
       * We need to restore the saved return address and return in
       * unprivileged thread mode.
       * unprivileged thread mode.
       */
       */
#ifdef CONFIG_LIB_SYSCALL

      case SYS_syscall_return:

```

Reply:
Ok, look like I am wrong.